backend/src/v2/driver/driver.go

```diff
@@ -1521,6 +1550,13 @@ func resolveInputParameter(
 		if !ok {
 			return nil, paramError(fmt.Errorf("parent DAG does not have input parameter %s", componentInput))
 		}
+
+		if v.GetStringValue() == EmptyOptionalPipelineInputValue {
+			// Return a known error in this case and let the caller choose if they can handle an empty optional pipeline
+			// input value.
+			return nil, ErrOptionalPipelineInputValue
+		}
```

Comment:
I don't understand why previously kfp decided to handle this in the launcher, to me it seems inputs should be resolved and submitted as part of executor input, so even component inputs should be handled at driver level

if we did that, we could forego the need for a sentinel right? we can just check if it is unset, and if component input definition provides a default and handle it right here?